build/migrate-wsl-from-v020.ps1

```diff
@@ -0,0 +1,22 @@
+# PowerShell script to ensure that if there's a 
+# This script returns 0 if a successful migration was carried out, 100 if no migration is necessary
+
+$parentKey = "HKCU:\SOFTWARE\Microsoft\Windows\CurrentVersion\Lxss"
+$guids = Get-ChildItem -Path $parentKey | Select-Object -ExpandProperty Name
+foreach ($guid in $guids) {
+   $prop = Get-ItemProperty -Path ("Registry::" + $guid)
+   if ($prop.DistributionName -eq "k3s") {
+     # Need to shut it down. Couldn't get this to work:
+     # wsl --list --running | select-string -Pattern k3s
+     # So just shut it down. We're installing, we shouldn't be running rancher-desktop anyway.
+
+     wsl --terminate k3s
+     
+     Set-ItemProperty -Path ("Registry::" + $guid) -Name DistributionName -Value "rancher-desktop"
```

Comment:
What is `Registry::`? I would have expected `$parentKey` here (in which case I'd have expected a `Join-Path` invocation).

Reply:
No, if you don't indicate this is a registry path the cmdlet interprets `HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\CurrentVersion\Lxss` as a subdirectory of `C:\Users\User` which of course doesn't exist. Powershell tries to treat registry hives and subdirectories orthogonally, but the abstraction falls apart too often -- which is why this PR took me all day to get right, and not an hour or so.

I wouldn't be surprised if there is a PS way of getting this to work, but the docs weren't helpful. Over to Mook to find that way.

Reply:
I thought `$parentKey` starts with `HKCU:\` and therefore it's on the `HKCU` "drive" (registry hive), and that's why things worked?

I'm more confused about the magical `Registry::` string, really, since that doesn't mention any part of `SOFTWARE\Microsoft\Windows\CurrentVersion\Lxss` at all.

Reply:
Oh, I see, my mistake; `-ExpandProperty Name` gives out strings like `HKEY_CURRENT_USER\…` instead of what the table format displays… my mistake.

Maybe `-ExpandProperty PSPath` would be useful instead?

Reply:
Notice that I've traded `$parentKey` for its subkeys (in the `$guids` array). Each of these looks like `HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\CurrentVersion\Lxss\{<GUID>}` and the leading colon is dropped. The docs pointed to the `Registry::` header, and prepending that is more straightforward than converting `HKEY_CURRENT_USER\` to `HKCU:\`.